src/realm.h

```diff
@@ -366,12 +366,16 @@ typedef enum realm_property_flags {
 typedef struct realm_notification_token realm_notification_token_t;
 typedef struct realm_callback_token realm_callback_token_t;
 typedef struct realm_refresh_callback_token realm_refresh_callback_token_t;
+typedef struct realm_async_transaction_token realm_async_transaction_token_t;
 typedef struct realm_object_changes realm_object_changes_t;
 typedef struct realm_collection_changes realm_collection_changes_t;
 typedef void (*realm_on_object_change_func_t)(realm_userdata_t userdata, const realm_object_changes_t*);
 typedef void (*realm_on_collection_change_func_t)(realm_userdata_t userdata, const realm_collection_changes_t*);
 typedef void (*realm_on_realm_change_func_t)(realm_userdata_t userdata);
 typedef void (*realm_on_realm_refresh_func_t)(realm_userdata_t userdata);
+typedef void (*realm_on_realm_async_begin_transaction_func_t)(realm_t*, realm_userdata_t userdata);
```

Comment:
I suggest calling these something like `realm_async_begin_transaction_func_t` and `realm_async_commit_transaction_func_t`. The `realm_on_XYZ` pattern is for repeatable callbacks, like notifications, where async transactions have one-and-done completion callbacks. Or better yet, since they're only used in one place each, you can define the function pointer type inside the parameter type in the function declaration.

Reply:
Why are we passing a pointer to the Realm in these callbacks?